jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/comments/CaseCommentsViewImpl.java

```diff
@@ -0,0 +1,215 @@
+        addCommentView(editing, editActionLabel, commentId, author, commentText, commentAddedAt, "fa-comment-o", actions);
+    }
+
+    private void addCommentView(boolean editing, String editActionLabel, final String commentId, final String author,
+                                final String commentText, final Date commentAddedAt, final String iconType,
+                                final CaseCommentsPresenter.CaseCommentAction... actions) {
+        final CaseCommentItemView commentItemView = provider.get();
+        commentItemView.setCommentAuthor(author);
+        commentItemView.setCommentText(commentText);
+        commentItemView.setCommentAddedAt(DateUtils.getDateStr(commentAddedAt));
+
+        commentItemView.setIconType(iconType);
+        commentItemView.setEditMode(editing);
+
+        if (editActionLabel != null) {
+            for (CaseCommentsPresenter.CaseCommentAction action : actions) {
+                commentItemView.addAction(action);
+            }
+            if(editing) {
+                commentItemView.addUpdateCommentAction(new CaseCommentsPresenter.CaseCommentAction() {
+
+                    @Override
+                    public String label() {
+                        return translationService.format(SAVE);
+                    }
+
+                    @Override
+                    public void execute() {
+                        if (commentItemView.validateForm()) {
+                            presenter.updateCaseComment(commentItemView.getUpdatedComment(), commentId);
+                            commentItemView.setEditMode(false);
+                        }
+                    }
+                });
+            }
+            commentItemView.addAction(new CaseCommentsPresenter.CaseCommentAction() {
+
+                @Override
+                public String label() {
+                    if(editing) {
+                        return translationService.format(CANCEL);
+                    }
+                    return translationService.format(EDIT);
+
+                }
+
+                @Override
+                public void execute() {
+                    String cmmntId="";
+                    if(!editing){
+                        cmmntId = commentId ;
+                    }
+                    presenter.setCurrentUpdatedCommentId(cmmntId);
+                    commentItemView.setEditMode(!editing);
+                    presenter.refreshComments();
+                }
+            });
+        }
+        comments.appendChild(commentItemView.getElement());
+    }
+
+
+    @EventHandler("addCommentButton")
+    @SuppressWarnings("unsued")
+    public void addCommentButton(@ForEvent("click") final Event e) {
+        submitCommentAddition();
+    }
+
+    @EventHandler("comment-creation-input")
+    @SuppressWarnings("unsued")
+    public void addCommentPressingEnter(@ForEvent("keypress") final KeyboardEvent e) {
+        if ("Enter".equals(e.getKey())){
```

Comment:
@cristianonicolai Yeah. That being said you *might* want to use a "keyup" event instead. There's good description [here](http://stackoverflow.com/questions/3396754/onkeypress-vs-onkeyup-and-onkeydown) about the differences, but the main thing is that the "keyup" event only fires once if the user holds down the key, whereas "keypress" and "keydown" will fire many times.